### DISCUSSION TITLE
SB5.1 - OCI - Deploy/UCI/gql failed

### DISCUSSION THREAD

**Discussion by heungheung** [Sunbird-Ed/Community#373] [Installation]
the following error when deploy gql
```
TASK [sunbird-deploy : copy and template env file from swarm role] *************
failed: [localhost] (item=../../../../ansible/roles/stack-sunbird/templates/gql.env) => {
    "ansible_loop_var": "item",
    "changed": false,
    "item": "../../../../ansible/roles/stack-sunbird/templates/gql.env"
}

MSG:

AnsibleUndefinedVariable: 'hasura_graphql_admin_secret' is undefined
```
basically 
  https://github.com/project-sunbird/sunbird-devops/blob/release-5.1.0/ansible/roles/stack-sunbird/templates/gql.env#L3
is referring hasura_graphql_admin_secret but we couldn't find hasura_graphql_admin_secret has been defined anywhere. 

Should it be part of core inventory or group vars? What should this values be?

**Comment by heungheung** [comment]
deploy success if I add hasura_graphql_admin_secret to Core/secret - not sure if that is the right way or there will be another layer in between.

Side track, 
Q1. not just the user need urlencode, it applies to the DB password. Thus, better also consider
https://github.com/project-sunbird/sunbird-devops/blob/release-5.1.0/ansible/roles/stack-sunbird/templates/gql.env
DB password got special chars. 
Q2. what should be UCI_CORE_BASE_URL in https://github.com/project-sunbird/sunbird-devops/blob/release-5.1.0/ansible/roles/stack-sunbird/templates/gql.env#L18 and why it is HCed to  11.2.6.6/inbound-service

**Reply by beepdot** [reply]
@saiakhil46 Please check this and raise a PR for the templates if required.

**Reply by saiakhil46** [reply]
@heungheung we are adding these details in our template, we will share you the details shortly.

**Reply by saiakhil46** [reply]
@heungheung you can follow below instructions to generate the values
use password generator tool like https://passwordsgenerator.net/ and generate password with length 33
example: hDF5fh9QEaW4vFjx6E4CVPPtQm8FtqJZ9
`uci_api_admin_token: "hDF5fh9QEaW4vFjx6E4CVPPtQm8FtqJZ9"`

use password generator tool like https://passwordsgenerator.net/ and generate password with length 43
example: ZpfFgp75ncgs7w9rp96rMYzSFvr5T3H5QxkvsvQzmPM
`hasura_graphql_admin_secret: "ZpfFgp75ncgs7w9rp96rMYzSFvr5T3H5QxkvsvQzmPM"`

use password generator tool like https://passwordsgenerator.net/ and generate password with length 57
example: dZfGV6x2MLAsJTvbw87tZyhfqNgnawHz9LYrespT6WMJkmtLYLbmKVCRA
`fusionauth_service_admin_key: "dZfGV6x2MLAsJTvbw87tZyhfqNgnawHz9LYrespT6WMJkmtLYLbmKVCRA"`

use password generator tool like https://passwordsgenerator.net/ (include symbols) and generate password with length 17
example: ^26C~\3$T~A2xs6d#
`uci_encryption_key_string: "^26C~\3$T~A2xs6d#"`

Take the value of uci_encryption_key_string variable and use below command to get base64 encrypted string
command: echo -n "^26C~\3$T~A2xs6d#" | base64
`uci_encryption_key_base64: "XjI2Q35cMyRUfkEyeHM2ZCM="`

Give some prefered username
`uci_odk_user: admin`

Give some prefered password
`uci_odk_password: admin`

**Reply by saiakhil46** [reply]
@heungheung we are also updating our template, meanwhile you can follow above mentioned instructions

**Reply by chitranshu-keshav** [reply]
@heungheung Can you please mark this thread as answered if your issue is resolved

**Reply by heungheung** [reply]
@chitranshu-keshav not fully answered. As I've mentioned above
https://github.com/project-sunbird/sunbird-devops/blob/release-5.1.0/ansible/roles/stack-sunbird/templates/gql.env#L18
what should be 
UCI_CORE_BASE_URL=11.2.6.6/inbound-service
Should this value got templated? or it is ALWAYS 11.2.6.6?

**Reply by chitranshu-keshav** [reply]
@saiakhil46 Can you please get back on the above query

**Comment by saiakhil46** [comment]
@heungheung it should be like below.
`UCI_CORE_BASE_URL={{ private_ingressgateway_ip }}/inbound-service`

I will also update in private_repo template, meanwhile use above info
